src/client/caching/stores/ChannelStore.ts

```diff
@@ -0,0 +1,31 @@
+import { DataStore } from './base/DataStore';
+import { Channel } from '../structures/channels/Channel';
+import { extender } from '../../../util/Extender';
+
+import type { APIChannelData } from '@klasa/dapi-types';
+import type { Client } from '../../Client';
+
+export class ChannelStore extends DataStore<Channel> {
+
+	public constructor(client: Client) {
+		super(client, extender.get('Channel'));
+	}
+
+	/**
+	 * Adds a new structure to this DataStore
+	 * @param data The data packet to add
+	 * @param cache If the data should be cached
+	 */
+	// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+	// @ts-ignore
+	protected _add(data: APIChannelData, cache = true): Channel | null {
+		const existing = this.get(data.id);
+		// eslint-disable-next-line dot-notation
+		if (existing && existing.type === data.type) return existing['_patch'](data);
```

Comment:
_patch doesn't return the structure

Reply:
Ok so apparently it does, `_patch` shouldn't return the structure, it should be `void`